client/state/site-profiler/reducer.js

```diff
@@ -0,0 +1,38 @@
+import { combineReducers } from '@wordpress/data';
+import {
+	SITE_PROFILER_SET_REPORT,
+	SITE_PROFILER_SET_LAST_VISITED_REPORT,
+} from 'calypso/state/action-types';
+
+const reportReducer = ( state = {}, action ) => {
+	const { siteId, url, hash, pageId } = action;
+	switch ( action.type ) {
+		case SITE_PROFILER_SET_REPORT:
+			return {
+				...state,
+				[ siteId ]: {
+					...( state[ siteId ] || {} ),
```

Comment:
JS fact: you can actually use the spread operator on `undefined` or `null` values and they do the same thing as spreading an empty object 👍 

```suggestion
					...state[ siteId ],
```